src/services/AssetService.ts

```diff
@@ -21,34 +28,49 @@ export class Assets {
         this.idProperty = 'uuid';
     }
 
-    public get(uuid: string): Promise<any> {
-        const existing = this.objects[uuid];
+    public get(id: string): Promise<Asset> {
+        const existing = this.assets[id];
         if (existing) {
-            return new Promise<any>(resolve => {
+            return new Promise<Asset>(resolve => {
                 resolve(existing);
             });
         }
 
-        return new Promise<any>((resolve, reject) => {
-            const url = `${this.endpoint}?uuid=${uuid}`;
+        return new Promise<Asset>((resolve, reject) => {
+            const url = `${this.endpoint}?uuid=${id}`;
             axios
                 .get(url)
                 .then((response: AxiosResponse) => {
                     const ob = response.data.results[0];
+                    const asset = {
+                        id: ob.uuid,
+                        name: ob.name,
+                        type: this.assetType,
+                        content: ob.definition
```

Comment:
just curious, why might a flow asset also need its definition?

Reply:
this is the case of fetching by uuid and getting the full definition

Reply:
ah, doh, thanks